src/main/java/redis/clients/jedis/CacheJedis.java

```diff
@@ -0,0 +1,265 @@
+package redis.clients.jedis;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import redis.clients.jedis.util.SafeEncoder;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.SSLParameters;
+import javax.net.ssl.SSLSocketFactory;
+import java.net.URI;
+import java.util.List;
+
+import static redis.clients.jedis.Protocol.Command.CLIENT;
+
+public class CacheJedis extends Jedis {
+    final Jedis jedisSub = new Jedis(); // Jedis instance for Client tracking
+    final String clientId= String.valueOf(jedisSub.clientId());
```

Comment:
1. Space after clientId
2. private variable 
3. all these initializations will move to builder